ivy/functional/frontends/numpy/creation_routines/numerical_ranges.py

```diff
@@ -156,3 +156,25 @@ def __init__(self):
 
 
 ogrid = OGrid()
+
+
+@handle_numpy_dtype
+@to_ivy_arrays_and_back
+def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
+    start = ivy.array(start)
+    stop = ivy.array(stop)
+    n = len(start)
+    a = []
+    for i in range(n):
+        cr = ivy.log((stop[i] / start[i]), where=start[i] != 0 )/ (num - 1 if endpoint else num)
+        if dtype is None:
+            start, stop = promote_types_of_numpy_inputs(x1, x2)
+            dtype = start.dtype
+        x = ivy.linspace(0, cr * (num - 1 if endpoint else num), num, endpoint=endpoint, dtype=dtype, axis=axis)
+        x = ivy.exp(x)
+        x = start[i] * x
+        b = ivy.reshape(x, (num,1))
+        if endpoint:
+            b[-1] = stop[i]
+        a.append(b)
+    return ivy.concatenate(a, axis=1)
```

Comment:
Quite a few comments here, firstly the promotion should be written like this: `start, stop = promote_types_of_numpy_inputs(start, stop)` and `promote_types_of_numpy_inputs` should be imported. That should be the first thing you do it will also convert start and stop to arrays so those conversions can be removed as well. The `if dtype is None` statement in unecessary. Also the for loop in unecessary and will slow things down A LOT. The main functions here are `ivy.log`, `ivy.linspace` and `ivy.exp`, all those are broadcastable so a for loop in unecessary remove it and keep the rest except the indexing variables `i` and `j` and the last 5 lines. Just return x after line 175.